components/calendar/src/chinese_based.rs

```diff
@@ -264,12 +267,67 @@ impl<C: ChineseBased + CalendarArithmetic> ChineseBasedDateInner<C> {
 
     /// Returns true if the fixed date given is in a leap year, false otherwise
     pub(crate) fn fixed_date_is_in_leap_year(date: RataDie) -> bool {
-        let prev_new_year = Self::new_year_on_or_before_fixed_date(date);
-        let next_new_year = Self::new_year_on_or_before_fixed_date(prev_new_year + 400);
+        let (prev_new_year, next_new_year) = Self::new_years_bounding_fixed_date(date);
         let difference = next_new_year - prev_new_year;
         difference > 365
     }
 
+    /// Get the (prev_new_year, next_new_year) RataDies bounding the given date.
+    fn new_years_bounding_fixed_date(date: RataDie) -> (RataDie, RataDie) {
+        let new_years = Self::new_years_bounding_date_in_sui(date);
+        if date >= new_years.0 {
+            new_years
+        } else {
+            Self::new_years_bounding_date_in_sui(date - 180)
+        }
+    }
+
+    /// Get the (prev_new_year, next_new_year) RataDies in the sui containing the given date.
+    fn new_years_bounding_date_in_sui(date: RataDie) -> (RataDie, RataDie) {
```

Comment:
Suggestion: try to reduce this duplicated code by passing in optional pre-computed values to the existing functions

Reply:
Done